contract_sale_generation/models/abstract_contract.py

```diff
@@ -2,16 +2,16 @@
 # Copyright 2017 Angel Moya <angel.moya@pesol.es>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ContractAbstractContract(models.AbstractModel):
     _inherit = "contract.abstract.contract"
 
-    type = fields.Selection(
-        [("invoice", "Invoice"), ("sale", "Sale")],
-        string="Type",
-        default="invoice",
-        required=True,
-    )
     sale_autoconfirm = fields.Boolean(string="Sale Autoconfirm")
+
+    @api.model
+    def _selection_generation_type(self):
+        res = super()._selection_generation_type()
+        res.append(("sale", "Sale"))
```

Comment:
Is this going to be translatable? Or do you need to add `_()`?

Reply:
The Selection field behaviour does not need it